### PR TITLE
[CMAKE][LWIP] Fix build with newer CMake versions

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Fix build with newer CMake versions

### CODE REVIEW

File: drivers/network/tcpip/lwip/src/Filelists.cmake [hunk 1]
```diff
@@ -262,19 +262,19 @@ set(DOXYGEN_IN  ${LWIP_DIR}/doc/doxygen/lwip.Doxyfile.cmake.in)
...
+#
+#    add_custom_target(lwipdocs
+#        COMMAND ${CMAKE_COMMAND} -E remove_directory ${DOXYGEN_DIR}/${DOXYGEN_OUTPUT_DIR}/html
+#        COMMAND ${DOXYGEN_EXECUTABLE} ${DOXYGEN_OUT}
+#        WORKING_DIRECTORY ${DOXYGEN_DIR}
+#        COMMENT "Generating API documentation with Doxygen"
+#        VERBATIM)
+#else (DOXYGEN_FOUND)
+#    message(STATUS "Doxygen needs to be installed to generate the doxygen documentation")
+#endif (DOXYGEN_FOUND)
```
Comment (binarymaster) [inline drivers/network/tcpip/lwip/src/Filelists.cmake:277]: Well, this looks like a hack. And it must be breaking Doxygen in some way.

Comment (learn-more) [reply drivers/network/tcpip/lwip/src/Filelists.cmake:277]: I don't think that we have to maintain doxygen stuff for a third-party component?

Comment (binarymaster) [reply drivers/network/tcpip/lwip/src/Filelists.cmake:277]: It might be important for the third-party components that are tightly integrated with ours.

Comment (learn-more) [reply drivers/network/tcpip/lwip/src/Filelists.cmake:277]: That should not need a target per library?
Which is what is being added here.

**Review by learn-more — APPROVED** [review]

**Comment by hpoussin** [comment]
Which cmake version are you using ? Which error message are you trying to fix ?

**Review by HBelusca — APPROVED** [review]

**Comment by Vince1171** [comment]
> Which cmake version are you using ? Which error message are you trying to fix ?

RosBE is broken on Linux since d6eebaa   
this PR fixes it.
any ninja/configure command results in this:

```console
(RosBE) vince@vince-Mint:~/Documents/sources/reactos/output-MinGW-i386$ ninja bootcd
[0/1] Re-running CMake...
Cross-compiling on non-msvc, no special host-tools cmake command
-- Build type: Debug
CMake Warning at CMakeLists.txt:223 (message):
  -- Disabling precompiled headers on GCC by default CORE-17108.


-- Selected localization: all
CMake Error at sdk/cmake/CMakeMacros.cmake:470 (add_custom_target):
  The target name "Doxygen::dot_clean" is reserved or not valid for certain
  CMake features, such as generator expressions, and may result in undefined
  behavior.
Call Stack (most recent call first):
  sdk/cmake/CMakeMacros.cmake:489 (add_clean_target)
  /usr/RosBE/share/cmake-3.17/Modules/FindDoxygen.cmake:543 (add_executable)
  /usr/RosBE/share/cmake-3.17/Modules/FindDoxygen.cmake:614 (_Doxygen_find_dot)
  drivers/network/tcpip/lwip/src/Filelists.cmake:265 (find_package)
  drivers/network/tcpip/CMakeLists.txt:7 (include)


-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Doxygen needs to be installed to generate the doxygen documentation
-- user32_vista has no base address
-- Configuring incomplete, errors occurred!
See also "/home/vince/Documents/sources/reactos/output-MinGW-i386/CMakeFiles/CMakeOutput.log".
FAILED: build.ninja 
/usr/RosBE/bin/cmake --regenerate-during-build -S/home/vince/Documents/sources/reactos -B/home/vince/Documents/sources/reactos/output-MinGW-i386
ninja: error: rebuilding 'build.ninja': subcommand failed
```

**Comment by tkreuzer** [comment]
> Which cmake version are you using ? Which error message are you trying to fix ?

I'm using cmake 3.27.4 for MSVC builds. Same error as commented above.

File: drivers/network/tcpip/lwip/src/Filelists.cmake (hunk 1 above)
Comment (tkreuzer) [reply drivers/network/tcpip/lwip/src/Filelists.cmake:277]: > Well, this looks like a hack. And it must be breaking Doxygen in some way.

Yes, it's a "hack" in the sense that it modifies 3rd party code, which we do in many places. We don't need this here, so there is no reason to keep it.

**Pull request merged**